src/ssh/cockpitsshrelay.c

```diff
@@ -1374,12 +1375,15 @@ cockpit_ssh_connect (CockpitSshData *data,
     }
 
   g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_HOST, host) == 0);
-#if LIBSSH_085
-    g_warn_if_fail (ssh_options_parse_config (data->session, NULL) == 0);
-#endif
-  g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_USER, data->username) == 0);
   if (port != 0)
     g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_PORT, &port) == 0);
+#if LIBSSH_085
+  g_warn_if_fail (ssh_options_parse_config (data->session, NULL) == 0);
+#endif
```

Comment:
Is that change relevant? I. e. setting the port before parsing the config?

Reply:
whitespace fix ._.

Reply:
The indentation change is fine, I'm just trying to understand why you swapped setting the port and parsing the config.

Reply:
Oh, so in the port case if the user specifies the port in the url, I think we'd wanna use that one over the one in the config. If the user specifies `user@host:1234` as a url those need to take precedent.

Which reminds me, adding `needswork` because I wanna make sure the user@ works.